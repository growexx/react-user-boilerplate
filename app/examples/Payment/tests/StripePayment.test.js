import React from 'react';
import { render, fireEvent } from 'react-testing-library';
import request from 'utils/request';
import StripePayment from '../StripePayment';

jest.mock('utils/request');
const props = {
  token: 'abc1234',
  requestPaymentStripe: () => {},
};
describe('<StripePayment />', () => {
  it('should render with token', () => {
    const {
      container: { firstChild },
    } = render(<StripePayment {...props} />);
    expect(firstChild).toMatchSnapshot();
  });
  it('should submit form called', () => {
    request.mockImplementationOnce(() =>
      Promise.resolve({ data: 'testValue' }),
    );
    const { getByTestId } = render(<StripePayment {...props} />);
    const button = getByTestId('stripePayBtn');
    fireEvent.click(button);
  });
});
