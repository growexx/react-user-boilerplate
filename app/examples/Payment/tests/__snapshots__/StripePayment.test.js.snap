// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StripePayment /> should render with token 1`] = `
<form>
  <div />
  <button
    data-testid="stripePayBtn"
    disabled=""
    style="margin-top: 50px;"
    type="submit"
  >
    Pay
  </button>
</form>
`;
